ha/core/controllers/pcs/cluster_controller.py

```diff
@@ -367,3 +367,23 @@ def create_cluster(self, name: str, user: str, secret: str, nodeid: str) -> dict
                 raise ClusterManagerError("Cluster is not started.")
         except Exception as e:
             raise ClusterManagerError(f"Failed to create cluster. Error: {e}")
+
+    @controller_error_handler
+    def destroy_cluster(self, retry_index: int = 0):
+        if retry_index < const.CLUSTER_RETRY_COUNT and not self._is_pcs_cluster_running():
+            Log.warning('Cluster is not running, safe to destroy the cluster')
+            output = self._execute.run_cmd(const.PCS_CLUSTER_DESTROY)
+            Log.error(f"Cluster destroyed. Output: {output}")
+            return
+        elif retry_index == 0:
+            cluster_stop_response = self.stop()
```

Comment:
Should we keep stop and destroy separate ?

@ajaysrivas @indrajitzagade

Reply:
This is fine. At first attempt, stop should be used and if it's not successful then kill+destroy can be used.

Reply:
Sure. Done.